### PR TITLE
SIP-365: Remove Core Fee Switch

### PR DESCRIPTION
**Pull request by JaredBorders**
Remove the core fee capture mechanism (implemented per [SIP-319](https://sips.synthetix.io/sips/sip-319/)) and associated logic.

## Description
* Remove fee logic from: `mintUsd`, `burnUsd`, `depositMarketUsd`, `withdrawMarketUsd`
* Remove tests asserting behavior relating to: `mintUsd`, `burnUsd`, `depositMarketUsd`, `withdrawMarketUsd`
* Remove the following events: `IssuanceFeePaid`, `MarketSystemFeePaid` @Tburm
* Update Core MarketManagerModule and respective Interface
* Update Core IssueUSDModule and respective interface
* Update test utility `MockMarket.sol`: update `sellSynth`, `buySynth`

## Related issue(s)
https://sips.synthetix.io/sips/sip-365/

## Motivation and Context
https://sips.synthetix.io/sips/sip-365/

### CODE REVIEW

File: protocol/synthetix/test/integration/modules/core/MarketManagerModule.test.ts [hunk 1]
```diff
@@ -289,6 +209,7 @@ describe('MarketManagerModule', function () {
 
       describe('withdraw some from the market', async () => {
         before(withdrawRestore);
+
```
Comment (noisekit) [inline protocol/synthetix/test/integration/modules/core/MarketManagerModule.test.ts:212]: I have pushed local branch `sip-365-remove-core-fee-switch` to the repo so CI will actually run. Will keep it up to date, keep an eye of failed workflows - they are legit now.

Comment (noisekit) [reply protocol/synthetix/test/integration/modules/core/MarketManagerModule.test.ts:212]: image

**Comment by Tburm** [comment]
No issues here, we aren't currently using those two events for anything.

File: protocol/synthetix/test/integration/modules/core/MarketManagerModule.test.ts (hunk 1 above)
Comment (noisekit) [reply protocol/synthetix/test/integration/modules/core/MarketManagerModule.test.ts]: Fixed.

**Comment by JaredBorders** [comment]
@noisekit not a fan of my emojis? haha

**Pull request merged**